notebook2/notebook/notebook.py

```diff
@@ -315,6 +347,51 @@ def wait_websocket(ws):
     log.info(f'notification sent to user for {svc_name} {pod_name}')
 
 
+@app.route('/auth0-callback')
+def auth0_callback():
+    # https://github.com/auth0-samples/auth0-python-web-app/commit/d048d6497caa714c52e8411a5f37500787e37305
+    auth0.authorize_access_token()
+
+    userinfo = auth0.get('userinfo').json()
+
+    email = userinfo['email']
+    workshop_password = session['workshop_password']
+    del session['workshop_password']
+
+    if AUTHORIZED_USERS.get(email) is None and workshop_password != PASSWORD:
+        return redirect(flask.url_for('login_page', unauthorized = True))
+
+    session['user'] = {
+        'user_id': userinfo['sub'],
+        'name': userinfo['name'],
+        'email': email,
+        'picture': userinfo['picture'],
+    }
+
+    return redirect('/')
+
+
+@app.route('/login', methods=['GET'])
+def login_page():
+    return render_template('login.html', unauthorized = request.args.get('unauthorized'))
+
+
+@app.route('/login', methods=['POST'])
+def login_auth0():
+    # FIXME ?: Could be placed outside route
```

Comment:
doesn't much matter to me. I tend to avoid optimizing my code until necessary.

Reply:
Either way, we should make a decision and remove FIXME.

Reply:
Yeah I really don’t like unnecessary assignments/regeneration of a constant right value, but I wanted your opinion.

I’ll remove